tests/Integration/CacheControl/CacheControlDirectiveTest.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+namespace Tests\Integration\CacheControl;
+
+use Tests\DBTestCase;
+use Tests\Utils\Models\Post;
+use Tests\Utils\Models\Task;
+use Tests\Utils\Models\Team;
+use Tests\Utils\Models\User;
+
+final class CacheControlDirectiveTest extends DBTestCase
+{
+    public function testDefaultAppResponseHeader(): void
+    {
+        $this->mockResolver([
+            'id' => 1,
+            'name' => 'foobar',
+        ]);
+
+        $this->schema /** @lang GraphQL */ = '
+        type User {
+            id: ID!
+            name: String
+        }
+
+        type Query {
+            user: User @mock
+        }
+        ';
+
+        $this->graphQL(/** @lang GraphQL */ '
+        {
+            user {
+                name
+            }
+        }
+        ')->assertHeader('Cache-Control', 'no-cache, private');
```

Comment:
I am not sure if this is the desired default behaviour, the Apollo docs say https://www.apollographql.com/docs/apollo-server/performance/caching/#calculating-cache-behavior

> The response's scope is PRIVATE if any field's scope is PRIVATE.

That would mean that fields without directives default to `PUBLIC` - that seems to be in line with the default in `@cacheControl`.

In any case, the behaviour is currently inconsistent, as illustrated by the failing test `testRootScalar` I added. Querying the root field `value` by itself defaults the cache to `private`, but adding another field makes it `public` - that does not make sense.

Reply:
You are mixing two things here:

1. The app default behavior was `no-cache, private` without any `@cacheControl`, this probably come from symfony or laravel. I code it in this way if there is no `@cacheControl` don't change anything.
2. When u add any `@cacheControl` then everything is calculated based on this directive, and the default for this directive is public unless u mention the private, thus the root field `value` is also `public` here.

Apollo didn't mention anything about the root field default scope.

Reply:
We can also do one of these:
- Make the default applications response public. 
- Make the route fields default scope public and mention it in doc.

Reply:
I think we should make an informed decision about the default scope for root fields. Is it a potential security risk to expose arbitrary fields with public scope? I would guess so, since a public response could be cached by proxy servers. If it holds private data, it could be accidentally shared with other users. I think they should default to private. Their default `max-age` is 0 anyway, so they won't be cached unless marked with `@cacheControl`.

Since Lighthouse currently follows the application default and this feature would override that, we should make it entirely optional. Let's do that by omitting the service provider from `composer.json` and let users add it explicitly.

Reply:
I made root fields default scope private anyway. Now if someone wants to use this, he must define `@cacheControl` for the root fields.

Reply:
> I made root fields default scope private anyway.

I understand this was your intention, but it does not behave that way - as illustrated by https://github.com/nuwave/lighthouse/pull/2136#discussion_r866679871